docs/csharp/tutorials/upgrade-to-nullable-references.md

```diff
@@ -0,0 +1,153 @@
+
+## Warnings help discover original design intent
+
+There are two classes that generate multiple warnings. Start with the `NewsStoryViewModel` class. Remove the `NullableContextOptions` element from both csproj files so that you can limit the scope of warnings to the sections of code you're working with. Open the `NewsStoryViewModel.cs` and add the following directives to enable the nullable annotation context for the `NewsStoryViewModel` and restore it following that class definition:
+
+```csharp
+#nullable enable
+public class NewsStoryViewModel
+{
+    public DateTimeOffset Published { get; set; }
+    public string Title { get; set; }
+    public string Uri { get; set; }
+}
+#nullable restore
+```
+
+These two directives help you focus your migration efforts. The nullable warnings are generated for the area of code you're actively working on. You'll leave them on until you're ready to turn on the warnings for the entire project. You should use the `restore` rather than `disable` value so that you don't accidentally disable the context later when you've turned on nullable annotations for the entire project.
+
+The `NewsStoryViewModel` class is a data transfer object (DTO) and two of the properties are read/write strings:
+
+[!code-csharp[InitialViewModel](../../../samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/ViewModels/NewsStoryViewModel.cs#StarterViewModel)]
+
+These two properties cause `CS8618`, "Non-nullable property is uninitialized". That's clear enough: both `string` properties have the default value of `null` when a `NewsStoryViewModel` is constructed. What's important to discover is how `NewsStoryViewModel` objects are constructed. Looking at this class, you can't tell if the `null` value is part of the design, or if these objects are set to non-null values whenever one is created. The news stories are created in the `GetNews` method of the `NewsService` class:
+
+[!code-csharp[StarterCreateNewsItem](../../../samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/Services/NewsService.cs#CreateNewsItem)]
+
+There's quite a bit going on in the preceding block of code. This application uses the [AutoMapper](http://automapper.org/) NuGet package to construct a news item from an `ISyndicationItem`. You've discovered that the news story items are constructed and the properties are set in that one statement. That means the design for the `NewsStoryViewModel` indicates that these properties should never have the `null` value. These properties should be **nonnullable reference types**. That best expresses the original design intent. In fact, any `NewsStoryViewModel` *is* correctly instantiated with non-null values. That makes the following initialization code a valid fix:
```

Comment:
This was an interesting part of the NDC talk for me - I didn't want to interrupt, but did want to mention it later.

Is AutoMapper configured to guarantee that it will populate all values? What about if the `ISyndicationItem` has some null values, possibly because they're missing in the feed itself?
That's probably useful to consider for an article in its own right...

Reply:
Yes, this is a subject for an entire article. There's two interesting cases: 

1. where your code calls a 3rd party library that is nullable oblivious.
1. where a nullable oblivious library invokes your code. (Automapper and DI containers are good examples).